paddle/fluid/framework/details/build_strategy.cc

```diff
@@ -47,212 +47,195 @@ class ParallelExecutorPassBuilder : public ir::PassBuilder {
+    context->endpoints_ = strategy_.trainers_endpoints_;
+    context->trainer_id_ = strategy_.trainer_id_;
+    PADDLE_ENFORCE_GE(strategy_.trainer_id_, 0, "trainer_id_ >= 0");
+    if (strategy_.trainer_id_ > 0 && strategy_.trainers_endpoints_.size() > 0) {
+      PADDLE_ENFORCE_LT(static_cast<size_t>(strategy_.trainer_id_),
+                        strategy_.trainers_endpoints_.size(),
+                        "trainer_id_ < endpoints_ size");
     }
-
-    // Verify that the graph is correct for multi-device executor.
-    VLOG(1) << "Add multi_devices_check_pass";
-    AppendPass("multi_devices_check_pass");
+    VLOG(1) << "CollectiveContext:" << context->String();
   }
 
   // Convert graph to run on multi-devices.
-  void AppendMultiDevPass(const BuildStrategy &strategy) {
+  void AppendMultiDevPass() {
     ir::Pass *multi_devices_pass = nullptr;
-
     if (strategy_.async_mode_) {
-      VLOG(1) << "Add async_multi_devices_pass";
       multi_devices_pass = AppendPass("async_multi_devices_pass").get();
     } else if (strategy_.is_distribution_) {
-      VLOG(1)
-          << "Add dist_multi_devices_pass, multi device parameter server mode";
       multi_devices_pass = AppendPass("dist_multi_devices_pass").get();
     } else {
-      if (strategy.reduce_ == BuildStrategy::ReduceStrategy::kAllReduce) {
-        VLOG(1) << "Add all_reduce_mode_multi_devices_pass";
-        multi_devices_pass =
-            AppendPass("all_reduce_mode_multi_devices_pass").get();
-      } else if (strategy.reduce_ == BuildStrategy::ReduceStrategy::kReduce) {
-        VLOG(1) << "Add reduce_mode_multi_devices_pass";
-        multi_devices_pass = AppendPass("reduce_mode_multi_devices_pass").get();
-      } else {
-        PADDLE_THROW("Unknown reduce strategy.");
+      switch (strategy_.reduce_) {
+        case BuildStrategy::ReduceStrategy::kAllReduce:
+          multi_devices_pass =
+              AppendPass("all_reduce_mode_multi_devices_pass").get();
+          break;
+        case BuildStrategy::ReduceStrategy::kReduce:
+          multi_devices_pass =
+              AppendPass("reduce_mode_multi_devices_pass").get();
+          break;
+        default:
+          PADDLE_THROW("Unknown reduce strategy.");
       }
     }
     multi_devices_pass->SetNotOwned<const BuildStrategy>("strategy",
                                                          &strategy_);
   }
 
+  void AppendPrintGraphPass(const std::string &pass_name,
+                            const std::string &debug_file_suffix) {
+    if (!strategy_.debug_graphviz_path_.empty()) {
+      auto viz_pass = AppendPass(pass_name);
+      const std::string graph_path = string::Sprintf(
+          "%s%s", strategy_.debug_graphviz_path_.c_str(), debug_file_suffix);
+      viz_pass->Set<std::string>(ir::kGraphvizPath,
+                                 new std::string(graph_path));
+    }
+  }
+
+  void AppendPassWithCheck(bool append_pass, const std::string &pass_name) {
+    if (append_pass) {
```

Comment:
The pass used for optimize graph is important info for user to debug and why delete the `VLOG(1)`?